final-project/src/tier1/main.py

```diff
@@ -47,55 +45,68 @@ def __init__(self, debug=False):
         # Moving setpoint time counter
         self.moving_setpoint_time_counter = 0.0
         self.moving_setpoint_time_counter_increment = 0.5
+        self.env.reset_idx([0])
 
     def choose_new_goal_position(self):
         """Chooses a new goal position"""
         # Choose a random index from the goal position options
         # self.goal_position = self.goal_position_options[
         #     torch.randint(0, self.goal_position_options.shape[0], (1,)).item()]
         # Just choose a coordinate along the x-axis
-        self.goal_position = torch.tensor([19.0, 0.0, 7.5],
+        self.goal_position = torch.tensor([19.0, 19.0, .2],
+        # self.goal_position = torch.tensor([13.0, 13.0, 0.2],
                                           dtype=torch.float32, device=self.device)
         if self.debug:
             print("New goal position: ", self.goal_position)
 
     def step(self, action):
+        
         # Capture starting relative position
         self.reward_function.dt_start = self.get_distance_from_moving_setpoint()
         if self.debug:
             print("dt_start: ", self.reward_function.dt_start)
+        
+        print("Start Position: ", self.get_current_position())
+        print("action: ", action)
+        
         # Get action bezier curve as sampled points
         num_samples = 5  # THIS MUST NEVER BE LESS THAN 3 OR THE NN WILL FAIL
         points = self.action_primitives.get_sampled_curve(action, num_samples=5)
         # Step through all points
         reset = False
         collision = False
-        for i_sample in range(0, num_samples):
+        for i_sample in range(4, num_samples):
             # Set command actions
             self.command_actions = torch.from_numpy(points[:, i_sample])
-            for _ in range(0, 15):
+            for _ in range(0, 1):
```

Comment:
If you're going to have these inner and outer loops just execute once, why not just remove them?

And are you having them execute only once because the control step per env step is now 100?

Reply:
Agreed, I am also a bit confused why we want to run the outer loop once

Reply:
This was a hack to set the moving location directly to the intermediate goal pointbased on what the action would yield as displayed in the figure.  We have an action item to pass in a dynamic list of intermediate points and to make the pre-control code perform the necessary loops.  So, this will go away in next work item.

Reply:
Please make a task or something so that we can track that it will be gone in the next PR.